deploy.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+
+set -e
+
+VERSION=`date "+%Y%m%dt%H%M%S"`
+LOGFILE="$VERSION.log"
+
+MIGRATE="--migrate"
+if [ "$GCLOUD_PROJECT" = "dartlang-pub-dev" ]; then
+  MIGRATE=""
+fi
+
+ANALYZER_HOST="https://""$VERSION""-dot-analyzer-dot-""$GCLOUD_PROJECT"".appspot.com"
+SEARCH_HOST="https://""$VERSION""-dot-search-dot-""$GCLOUD_PROJECT"".appspot.com"
+APP_HOST="https://""$VERSION""-dot-""$GCLOUD_PROJECT"".appspot.com"
+
+if [ "$1" = "all" ] || [ "$1" = "analyzer" ]; then
+  OLD_ANALYZER=`gcloud app versions list --service analyzer --format='value(id)'`
+
+  echo "Deploying analyzer ($VERSION)..."
+  echo "Deploying analyzer ($VERSION)..." >>$LOGFILE
+  gcloud app deploy analyzer.yaml --no-promote -v $VERSION -q >>$LOGFILE 2>&1
+
+  echo "Checking analyzer status on $ANALYZER_HOST ..."
+  echo "Checking analyzer status on $ANALYZER_HOST ..." >>$LOGFILE
+  curl "$ANALYZER_HOST""/debug" -s -f  >/dev/null
+
+  echo "Migrating traffic on analyzer..."
+  echo "Migrating traffic on analyzer..." >>$LOGFILE
+  gcloud app services set-traffic analyzer --splits $VERSION=1 $MIGRATE -q >>$LOGFILE 2>&1
+
+  if [ "$2" = "--delete-old" ]; then
+    if [ $OLD_ANALYZER == *$'\n'* ]; then
+      echo "More than one existing analyzer versions, exiting."
+      exit 1;
+    fi
+    if [ $OLD_ANALYZER = "" ]; then
+      echo "No existing analyzer version, exiting."
+      exit 1;
+    fi
+    echo "Delete old analyzer ($OLD_ANALYZER)..."
+    echo "Delete old analyzer ($OLD_ANALYZER)..." >>$LOGFILE
+    gcloud app versions delete --service analyzer $OLD_ANALYZER -q >>$LOGFILE 2>&1
+  fi
+fi
+
+if [ "$1" = "all" ] || [ "$1" = "search" ]; then
```

Comment:
There's a lot of duplicated code, it's hard to see what the difference is. You could use bash functions with arguments to remove the duplication.